### PR TITLE
End input variants and add textarea

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: design-system/atoms/input/adapters/default/default.variants.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    wrapper: "flex w-full items-stretch justify-start",
+    base: [
+      "transition-all",
+      "flex w-full items-stretch justify-start gap-md rounded-md border-1 border-border-primary bg-background-primary-alt effect-box-shadow-xs",
+      "data-[hover=true]:border-border-primary-hover hover:border-border-primary-hover",
+    ],
+    contentWrapper: "flex flex-col items-start justify-center text-foreground-tertiary",
+    inputWrapper: "flex-1",
+    input: [
+      "h-full w-full bg-transparent outline-none focus:outline-none",
```
Comment (haydencleary) [inline design-system/atoms/input/adapters/default/default.variants.ts:15]:
```suggestion
      "h-full w-full bg-transparent outline-none outline-none",
```

To also handle `focus-visible`

File: design-system/atoms/input/adapters/default/default.adapter.tsx [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+  }
+
+  function handleChanges(e: ChangeEvent<HTMLInputElement>) {
+    if (onChange) {
+      onChange(e);
+    }
+  }
+
+  return (
+    <div className={cn(slots.container(), classNames?.container)} {...attr}>
```
Comment (haydencleary) [inline design-system/atoms/input/adapters/default/default.adapter.tsx:197]: Is the spread of generic attributes necessary ?

File: design-system/atoms/input/adapters/default/default.adapter.tsx [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+            <input
+              name={name}
+              id={id}
+              ref={ref}
+              className={cn(slots.input(), classNames?.input)}
+              onFocus={onFocus}
+              onBlur={onBlur}
+              value={value}
+              placeholder={placeholder}
+              disabled={isDisabled || canInteract}
```
Comment (haydencleary) [inline design-system/atoms/input/adapters/default/default.adapter.tsx:212]:
```suggestion
              disabled={isDisabled || !canInteract}
```
This condition would make more sense, but by default the prop is falsy.

File: design-system/atoms/textarea/adapters/next-ui/next-ui.adapter.tsx [hunk 4]
```diff
@@ -1,11 +1,69 @@
...
 
 import { TextareaPort } from "../../textarea.types";
 import { TextareaNextUiVariants } from "./next-ui.variants";
 
+function LabelsContainer({
+  label,
+  description,
+  classNames,
+}: Pick<TextareaPort, "label" | "description" | "classNames">) {
+  if (!description && label) {
```
Comment (haydencleary) [inline design-system/atoms/textarea/adapters/next-ui/next-ui.adapter.tsx:18]:
```suggestion
  if (!description && !label) {
```

File: design-system/atoms/textarea/adapters/next-ui/next-ui.adapter.tsx [hunk 5]
```diff
@@ -1,11 +1,69 @@
...
+    return null;
+  }
+  return (
+    <div className="flex w-full flex-col">
+      {label ? (
+        <Typo as={"label"} size={"sm"} weight={"medium"} classNames={{ base: classNames?.label }}>
```
Comment (haydencleary) [inline design-system/atoms/textarea/adapters/next-ui/next-ui.adapter.tsx:24]: No `htmlFor`

File: design-system/atoms/input/adapters/default/default.adapter.tsx [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+const defaultSize = "md";
+
+function LabelsContainer({ label, description, classNames }: Pick<InputPort, "label" | "description" | "classNames">) {
+  if (!description && label) {
+    return null;
+  }
+  return (
+    <div className="flex w-full flex-col">
+      {label ? (
+        <Typo as={"label"} size={"sm"} weight={"medium"} classNames={{ base: classNames?.label }}>
```
Comment (haydencleary) [inline design-system/atoms/input/adapters/default/default.adapter.tsx:24]: No `htmlFor`

File: design-system/atoms/textarea/adapters/next-ui/next-ui.adapter.tsx [hunk 7]
```diff
@@ -1,11 +1,69 @@
...
+}
+
+function InfoContainer({ error, info, isError }: Pick<TextareaPort, "info" | "error" | "isError">) {
+  if (isError && error) {
+    return (
+      <div className="flex w-full flex-row items-center justify-start gap-sm text-foreground-error">
+        <Icon size={"sm"} {...(error.icon || {})} component={error?.icon?.component || CircleAlert} />
+        {error?.text ? (
+          <Typo as={"div"} size={"sm"} color={"error"}>
+            {error?.text}
```
Comment (haydencleary) [inline design-system/atoms/textarea/adapters/next-ui/next-ui.adapter.tsx:44]:
```suggestion
        <Icon size={"sm"} {...(error.icon || {})} component={error.icon?.component || CircleAlert} />
        {error.text ? (
          <Typo as={"div"} size={"sm"} color={"error"}>
            {error.text}
```

File: design-system/atoms/textarea/adapters/next-ui/next-ui.adapter.tsx [hunk 8]
```diff
@@ -1,11 +1,69 @@
...
+    );
+  }
+
+  if (info) {
+    return (
+      <div className="flex w-full flex-row items-center justify-start gap-sm text-foreground-secondary">
+        {info?.icon ? <Icon size={"sm"} {...info.icon} /> : null}
+        {info?.text ? (
+          <Typo as={"div"} size={"sm"} color={"secondary"}>
+            {info?.text}
```
Comment (haydencleary) [inline design-system/atoms/textarea/adapters/next-ui/next-ui.adapter.tsx:57]:
```suggestion
        {info.icon ? <Icon size={"sm"} {...info.icon} /> : null}
        {info.text ? (
          <Typo as={"div"} size={"sm"} color={"secondary"}>
            {info.text}
```

File: design-system/atoms/textarea/adapters/next-ui/next-ui.variants.ts [hunk 9]
```diff
@@ -2,29 +2,30 @@ import { tv } from "tailwind-variants";
...
     base: "flex h-fit flex-col items-start gap-2",
-    mainWrapper: "",
-    inputWrapper:
-      "border-container-stroke-separator text-text-1 group-data-[focus=true]:border-interactions-white-hover group-data-[hover=true]:border-interactions-white-hover rounded-md border !outline-none",
+    inputWrapper: [
+      "transition-all",
+      "flex w-full items-stretch justify-start gap-md rounded-md border-1 border-border-primary bg-background-primary-alt effect-box-shadow-xs",
+      "data-[hover=true]:border-border-primary-hover hover:border-border-primary-hover",
+      "px-xl py-lg",
+      "group-data-[focus=true]:border-border-active group-data-[focus=true]:effect-ring-brand-glued",
```
Comment (haydencleary) [inline design-system/atoms/textarea/adapters/next-ui/next-ui.variants.ts:12]:
```suggestion
      "group-data-[focus-visible=true]:border-border-active group-data-[focus-visible=true]:effect-ring-brand-glued",
```

File: design-system/atoms/input/adapters/default/default.adapter.tsx [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+          </div>
+          <EndContent endContent={endContent} endIcon={endIcon} button={button} {...variants} />
+        </div>
+        {isOuterButton && button ? (
+          <Button
+            {...button}
+            size={size}
+            classNames={{ base: "rounded-l-none !border-l-0 h-auto" }}
+            isDisabled={isDisabled}
+          />
```
Comment (haydencleary) [inline design-system/atoms/input/adapters/default/default.adapter.tsx:224]: I don't see this in the designs ?

File: design-system/atoms/input/adapters/default/default.adapter.tsx (hunk 2 above)
Comment (alexbeno) [reply design-system/atoms/input/adapters/default/default.adapter.tsx:197]: as the focus state can be use in other case in the design system like is done on checkbox and checkbox button the component should provide a way to apply a focus / hover state outside off the component

File: design-system/atoms/textarea/adapters/next-ui/next-ui.variants.ts (hunk 9 above)
Comment (alexbeno) [reply design-system/atoms/textarea/adapters/next-ui/next-ui.variants.ts:12]: focus-visible=true not working

File: design-system/atoms/input/adapters/default/default.adapter.tsx (hunk 10 above)
Comment (alexbeno) [reply design-system/atoms/input/adapters/default/default.adapter.tsx:224]: Capture d’écran 2024-09-05 à 17 04 44

File: design-system/atoms/textarea/adapters/next-ui/next-ui.variants.ts (hunk 9 above)
Comment (haydencleary) [reply design-system/atoms/textarea/adapters/next-ui/next-ui.variants.ts:12]: I can see it in the docs though https://nextui.org/docs/components/input#data-attributes 🤔

Comment (haydencleary) [reply design-system/atoms/textarea/adapters/next-ui/next-ui.variants.ts:12]: Ah but I'm probably mixing this up with button focus states that have unwanted behaviour, you can forget about this.

**Pull request merged**